src/ui/RoomSettings.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <QLabel>
+#include <QObject>
+#include <QPushButton>
+#include <QString>
+
+#include <mtx/events/guest_access.hpp>
+
+#include "CacheStructs.h"
+
+class TextField;
+
+/// Convenience class which connects events emmited from threads
+/// outside of main with the UI code.
+class ThreadProxy : public QObject
+{
+        Q_OBJECT
+
+signals:
+        void error(const QString &msg);
+        void nameEventSent(const QString &);
+        void topicEventSent(const QString &);
+        void stopLoading();
+};
+
+class EditModal : public QWidget
+{
+        Q_OBJECT
+
+public:
+        EditModal(const QString &roomId, QWidget *parent = nullptr);
+
+        void setFields(const QString &roomName, const QString &roomTopic);
+
+signals:
+        void nameChanged(const QString &roomName);
+        void topicChanged(const QString &topic);
+
+private slots:
+        void topicEventSent(const QString &topic);
+        void nameEventSent(const QString &name);
+        void error(const QString &msg);
+
+        void applyClicked();
+
+private:
+        QString roomId_;
+        QString initialName_;
+        QString initialTopic_;
+
+        QLabel *errorField_;
+
+        TextField *nameInput_;
+        TextField *topicInput_;
+
+        QPushButton *applyBtn_;
+        QPushButton *cancelBtn_;
+};
```

Comment:
You should not need that anymore, right?

Reply:
Yeah I guess the thread proxy is not really required so I can remove it, it was just for convenience anyway since I don't think it was needed before too.

As for the EditModal, I haven't added the name/topic editing yet so it's still uses the old edit modal.

Reply:
Oh, right, I mistook that for the room settings widget!

Reply:
So I don't need to remove the thread proxy class right?